api/lib/domain/events/handle-certification-partner.js

```diff
@@ -0,0 +1,84 @@
+const Badge = require('../models/Badge');
+const CertificationPartnerAcquisition = require('../models/CertificationPartnerAcquisition');
+const Promise = require('bluebird');
+const { MINIMUM_REPRODUCTIBILITY_RATE_TO_BE_CERTIFIED, MINIMUM_REPRODUCTIBILITY_RATE_TO_BE_TRUSTED } = require('../constants');
+
+const CERTIF_GREEN_ZONE = 'green_zone';
+const CERTIF_RED_ZONE = 'red_zone';
+
+const handleCertificationAcquisitionForPartner = async function({
+  assessmentCompletedEvent,
+  certificationScoringEvent,
+  domainTransaction,
+  badgeAcquisitionRepository,
+  certificationPartnerAcquisitionRepository,
+}) {
+  if (!assessmentCompletedEvent.isCertification) {
+    return;
+  }
+
+  const partnerCertifications = await _getAcquiredPartnerCertifications({
+    badgeAcquisitionRepository,
+    userId: assessmentCompletedEvent.userId,
+    certificationCourseId: certificationScoringEvent.certificationCourseId,
+    percentageCorrectAnswers: certificationScoringEvent.percentageCorrectAnswers,
+  });
+
+  await _saveResult({
+    partnerCertifications,
+    domainTransaction,
+    certificationPartnerAcquisitionRepository,
+  });
+};
+
+async function _saveResult({
+  partnerCertifications,
+  domainTransaction,
+  certificationPartnerAcquisitionRepository,
+}) {
+  return await Promise.map(partnerCertifications, (partnerCertification) => {
+    return certificationPartnerAcquisitionRepository.save(partnerCertification, domainTransaction);
+  });
+}
```

Comment:
Ici `Promise.map` retourne une promesse on a pas besoin de `await` au moment du return.

```suggestion
function _saveResult({
  partnerCertifications,
  domainTransaction,
  certificationPartnerAcquisitionRepository,
}) {
  return Promise.map(partnerCertifications, (partnerCertification) => {
    return certificationPartnerAcquisitionRepository.save(partnerCertification, domainTransaction);
  });
}
```

Reply:
Je crois que par convention chez PIX on met explicitement les async même s'il ne sont pas nécessaire. (A confirmer)

Reply:
Je laisse le async pour information de la signature mais pas le await qui effectivement est inutile